dvc/repo/metrics/show.py

```diff
@@ -80,71 +129,23 @@ def show(
     for branch in self.scm.brancher(
         all_branches=all_branches, all_tags=all_tags
     ):
-        astages = self.active_stages()
-        outs = [out for stage in astages for out in stage.outs]
-
-        if path:
-            outs = find_output_by_path(
-                self, path, outs=outs, recursive=recursive
-            )
-
-        metrics = filter(lambda o: o.metric, outs)
-        stages = set()
-        entries = []
-
-        for o in metrics:
-            if not typ and isinstance(o.metric, dict):
-                t = o.metric.get(o.PARAM_METRIC_TYPE, typ)
-                x = o.metric.get(o.PARAM_METRIC_XPATH, xpath)
-            else:
-                t = typ
-                x = xpath
-            entries.append((o.path, t, x))
-            stages.add(o.stage.path)
+        entries = _collect_metrics(self, path, recursive, typ, xpath)
 
         if path and not entries:
             if os.path.isdir(path):
                 logger.warning(
                     "Path '{path}' is a directory. "
```

Comment:
If we decide to keep `continue` let's add a branch name here?

Reply:
should we write any warning at all then? I would expect to see a similar warning when -R hits a file in some branch

Reply:
Removed the warning and re-did the find_outs_by_path to follow the initial logic.